### PR TITLE
[API] Refactor commands in ApiBundle

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Command/Promotion/GeneratePromotionCoupon.php [hunk 1]
```diff
@@ -17,16 +17,16 @@
...
-        private ?int $amount = null,
-        private ?\DateTimeInterface $expiresAt = null,
-        private ?int $usageLimit = null,
+        protected readonly string $promotionCode,
+        protected readonly ?string $prefix = null,
+        protected readonly ?int $codeLength = null,
+        protected readonly ?string $suffix = null,
+        protected readonly ?int $amount = null,
+        protected readonly ?\DateTimeInterface $expiresAt = null,
+        protected readonly ?int $usageLimit = null,
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ApiBundle/Command/Promotion/GeneratePromotionCoupon.php:29]:
```suggestion
        public readonly ?int $usageLimit = null,
```
?

File: src/Sylius/Bundle/ApiBundle/Command/Account/ResetPassword.php [hunk 2]
```diff
@@ -13,12 +13,15 @@
...
+#[TokenAware]
 class ResetPassword
 {
     public function __construct(
-        public string $token,
-        public ?string $newPassword = null,
-        public ?string $confirmNewPassword = null,
+        public readonly string $token,
+        public readonly string $newPassword,
+        public readonly string $confirmNewPassword,
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ApiBundle/Command/Account/ResetPassword.php]:
```suggestion
        #[\SensitiveParameter]
        public readonly string $newPassword,
        #[\SensitiveParameter]
        public readonly string $confirmNewPassword,
```
Same w/ the one in core

**Review by Wojdylak — DISMISSED** [review]

File: src/Sylius/Bundle/ApiBundle/Command/Promotion/GeneratePromotionCoupon.php (hunk 1 above)
Comment (GSadee) [reply src/Sylius/Bundle/ApiBundle/Command/Promotion/GeneratePromotionCoupon.php:29]: This one implements an interface with public getters from Promotion component, so the `protected` access modifiers would be enough IMO.

**Review by Wojdylak — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/shop/CustomerPassword.xml [hunk 3]
```diff
@@ -58,11 +58,12 @@
             <operation
                 name="sylius_api_shop_customer_patch_reset_password"
                 class="ApiPlatform\Metadata\Patch"
+                uriTemplate="/shop/customers/reset-password/{token}"
+                messenger="input"
                 input="Sylius\Bundle\ApiBundle\Command\Account\ResetPassword"
                 output="false"
+                read="false"
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/shop/CustomerPassword.xml:65]: Do we still need the `output` node?

**Review by NoResponseMate — APPROVED** [review]

**Comment by NoResponseMate** [comment]
Thank you, @GSadee!

**Pull request merged**